src/Process/FeatureProcessing.js

```diff
@@ -23,105 +23,86 @@ function applyOffset(obj, offset) {
+        }
 
-            const features = node.children.filter(n => n.layer == layer.id);
-            if (features.length > 0) {
-                return features;
-            }
+        const features = node.children.filter(n => n.layer == layer.id);
+        if (features.length > 0) {
+            return features;
+        }
 
-            if (!layer.tileInsideLimit(node, layer)) {
-                return;
-            }
+        if (!layer.tileInsideLimit(node, layer)) {
+            return;
+        }
 
-            if (node.layerUpdateState[layer.id] === undefined) {
-                node.layerUpdateState[layer.id] = new LayerUpdateState();
-            }
+        if (node.layerUpdateState[layer.id] === undefined) {
+            node.layerUpdateState[layer.id] = new LayerUpdateState();
+        }
 
-            const ts = Date.now();
+        const ts = Date.now();
 
-            if (!node.layerUpdateState[layer.id].canTryUpdate(ts)) {
-                return;
-            }
+        if (!node.layerUpdateState[layer.id].canTryUpdate(ts)) {
+            return;
+        }
 
-            node.layerUpdateState[layer.id].newTry();
+        node.layerUpdateState[layer.id].newTry();
 
-            const command = {
-                layer,
-                view: context.view,
-                threejsLayer: layer.threejsLayer,
-                requester: node,
-            };
+        const command = {
+            layer,
+            view: context.view,
+            threejsLayer: layer.threejsLayer,
+            requester: node,
+        };
 
-            context.scheduler.execute(command).then((result) => {
-                // if request return empty json, WFS_Provider.getFeatures return undefined
-                if (result) {
-                    node.layerUpdateState[layer.id].success();
-                    if (!node.parent) {
-                        ObjectRemovalHelper.removeChildrenAndCleanupRecursively(layer.id, result);
-                        return;
-                    }
-                    // We don't use node.matrixWorld here, because feature coordinates are
-                    // expressed in crs coordinates (which may be different than world coordinates,
-                    // if node's layer is attached to an Object with a non-identity transformation)
-                    const tmp = node.extent.center().as(context.view.referenceCrs).xyz().negate();
-                    applyOffset(result, tmp);
-                    if (colorFunction) {
-                        colorFunction(layer, node, result);
+        context.scheduler.execute(command).then((result) => {
+            // if request return empty json, WFS_Provider.getFeatures return undefined
+            if (result) {
+                // call onMeshCreated callback if needed
+                if (layer.onMeshCreated) {
+                    let mesh = result;
+                    if (result instanceof THREE.Group) {
+                        mesh = result.children[0];
```

Comment:
I don't think this is a good idea because the user won't be able to access to the other children.
You should call `onMeshCreated(result)` (and document *somewhere* what type of objects the user can expect)